src/jit/lower.h

```diff
@@ -153,6 +186,15 @@ class Lowering : public Phase
         return new (comp, GT_LEA) GenTreeAddrMode(resultType, base, index, 0, 0);
     }
 
+    unsigned ReplaceWithLclVar(LIR::Use& use, unsigned tempNum = BAD_VAR_NUM)
+    {
+        GenTree* oldUseNode = use.Def();
+        unsigned newLclNum  = use.ReplaceWithLclVar(comp, m_block->getBBWeight(comp), tempNum);
```

Comment:
Nit: it is probably worth eliding the replacement to avoid introducing a copy if `oldUseNode` is already a LclVar use.